src/AbstractCsv.php

```diff
@@ -259,6 +275,7 @@ protected function newInstance($class_name, $open_mode)
         $csv->input_bom    = $this->input_bom;
         $csv->output_bom   = $this->output_bom;
         $csv->newline      = $this->newline;
+        $csv->strip_bom    = $this->strip_bom;
```

Comment:
I believe the `strip_bom` property should not be attached to `AbstractCsv` class. This property should be only usable on the `Reader` class. 

IMHO to make the code as much flexible as possible you should add the strip bom capability directly in the `Modifier/QueryFilter.php` trait and reset the status to false between calls. 

**Pros:**
- the BOM stripping is restricted to the `Reader` class
- the BOM stripping can be set per queries (so stripping is only made when needed).

**Cons:**
- the BOM stripping can be set per queries. The dev must not forget to use it

Secondly keep in mind that the `fetchAssoc` method can be affected if the `strip_bom` is set to `true`  and the column names are taken from the first row (which is the default behavior).